packages/amplify-category-hosting/index.js

```diff
@@ -65,11 +65,11 @@
       // the user is hosting from S3 without CloudFront distribution and should throw an error
       if (service === 'S3AndCloudFront') {
         const cfnOutput = context.exeInfo?.amplifyMeta?.hosting?.S3AndCloudFront?.output;
-        if (typeof cfnOutput === 'object' && cfnOutput.CloudFrontSecureURL === undefined) {
+        if (typeof cfnOutput === 'object' && (cfnOutput === undefined || cfnOutput.CloudFrontSecureURL === undefined)) {
```

Comment:
I did some manual testing with a previous version of CLI and the CfnOutput was undefined so I modified the conditional here.

Reply:
@evcodes I know this has been merged already, but I'm just seeing it now. Functionality is not impacted, but `cfnOutput === undefined` will always yield `false`, because if it was undefined, the `typeof` check would have already short circuited the condition.

```
Welcome to Node.js v18.16.0.
Type ".help" for more information.
> let x = {}
> typeof x
'object'
> x = undefined
> typeof x
'undefined'
```